### ISSUE TITLE
Determine when re-write of RDFSource request/response bodies are required, and implement

### ISSUE THREAD

**Issue by birkland** [fcrepo4-labs/fcrepo-api-x#15]
As API-X provides a proxy endpoint for Fedora, repository resources exposed through API-X have different URIs than their fedora counterparts.  When relative URIs are used in requests and response, there is no issue.  Absolute URIs may need to be rewritten in request and response bodies.  This is akin to [mod_proxy_html](https://httpd.apache.org/docs/2.4/mod/mod_proxy_html.html), except for rdf.

Testing:  
- PATCH, POST, PUT, GET in supported media types (turtle, n-triples, sparql update) where absolute URIs are present
  - Measure latency/throughput impact of doing this

**Comment by acoburn** [comment]
This W3C document is relevant to this discussion: https://www.w3.org/TR/ct-guidelines/

**Comment by ajs6f** [comment]
We could use that for `fcrepo-transform`. Or maybe `fcrepo-transform` should factor into API-X, actually. Okay, different issue.

**Comment by birkland** [comment]
@acoburn Yes, though I'm thinking that many (most?) scenarios will resemble what [RFC3040](https://www.ietf.org/rfc/rfc3040.txt) describes as a 'surrogate'.  In particular

> surrogate
>       A gateway co-located with an origin server, or at a different
>       point in the network, delegated the authority to operate on behalf
>       of, and typically working in close co-operation with, one or more
>       origin servers.
> ...
> Where close co-operation between origin servers and surrogates
>       exists, this enables modifications of some protocol requirements,
>       including the Cache-Control directives in [1].  Such modifications
>       have yet to be fully specified.

We should also look at contemporary implementations of API gateways to see what best practices are in this regard.
